src/com/sfuentes/MedianAndSmallestTest.java

```diff
@@ -0,0 +1,110 @@
+package com.sfuentes;
+
+import java.util.Arrays;
+
+public class MedianAndSmallestTest {
+  public static void main(String[] args) {
+
+    calculateSmallest_Decimal();
+    calculateSmallest_SameNumbers();
+    calculateSmallest_Zero();
+    calculateMedian_Decimal();
+    calculateMedian_SameNumbers();
+    calculateMedian_Zero();
+
+  }
+
+  // Smallest Tests
+  public static void calculateSmallest_Decimal() {
+    MedianAndSmallest smallest = new MedianAndSmallest();
+    double[] grades = {9.5, 7.6, 8.7, 10.8, 5.9, 9.1};
+    double result = smallest.calculateSmallest(grades);
+
+    System.out.println("Test case 1: Test with decimal numbers: \n\t" +
+        "Test numbers: " + Arrays.toString(grades) +
+        "\n\tExpected result: 5.9 \n\t" +
+        "Actual result: " + result);
+    if (result == 5.9)
+      System.out.println("\tSuccessful test.");
+    else
+      System.out.println("\tTest failed.");
+  }
+
+  public static void calculateSmallest_SameNumbers() {
+    MedianAndSmallest smallest = new MedianAndSmallest();
+    double[] grades = {9.5, 9.5, 9.5, 9.5, 9.5, 9.5};
+    double result = smallest.calculateSmallest(grades);
+
+    System.out.println("Test case 2: Test with the same numbers: \n\t" +
+        "Test numbers: " + Arrays.toString(grades) +
+        "\n\tExpected result: 9.5 \n\t" +
+        "Actual result: " + result);
+    if (result == 9.5)
+      System.out.println("\tSuccessful test.");
+    else
+      System.out.println("\tTest failed.");
+  }
+
+  public static void calculateSmallest_Zero() {
+    MedianAndSmallest smallest = new MedianAndSmallest();
+    double[] grades = {0.0, 0.0, 0.0, 0.0};
+    double result = smallest.calculateSmallest(grades);
+
+    System.out.println("Test case 3: Test with all zero number: \n\t" +
+        "Test numbers: " + Arrays.toString(grades) +
+        "\n\tExpected result: 0.0 \n\t" +
+        "Actual result: " + result);
+    if (result == 0.0)
+      System.out.println("\tSuccessful test.");
+    else
+      System.out.println("\tTest failed.");
+  }
+
+  // Medium Tests
+  public static void calculateMedian_Decimal() {
```

Comment:
Here in the tests you should change the name of the method you are testing since you rename it to calculateAverage